ols/app/models/config.py

```diff
@@ -581,6 +634,8 @@ def validate_yaml(self) -> None:
             self.reference_content.validate_yaml()
         if self.authentication_config:
             self.authentication_config.validate_yaml()
+        if self.tls_config:
+            self.tls_config.validate_yaml()
```

Comment:
Should we be validating TLS config if it is disabled?

Reply:
simply we disable TLS using dev_config, for that we cant know if dev_config.disable_tls is tru or false while initialiing/validating the model

Reply:
We can still do it by passing a disable_tls flag into OLSConfig.validate_yaml(). What's the UX we're looking for here? I'd imagine if I set disable_tls: true I won't need to generate cert/key.

Reply:
this will require changing the entire way we are validating config in the ols.utils.config
i thought about it multiple times

the only thing i reached to is that the OLSConfig should be aware of enablement/disablement of tls
and that can be done if we move the enable/disable entry to the ols_config instead of dev_config

Reply:
what you've done (checking at startup) is fine.  In general the goal is to fail as early as possible.  We want to fail before we get a request from a client, which you have achieved.

the key goal is that:
1) we are validating the configuration is complete/correct
2) we do it before the server starts accepting requests